### DISCUSSION TITLE
Running tests is beginning to become extremely time consuming

### DISCUSSION THREAD

**Discussion by LukeWood** [keras-team/keras-cv#673] [General]
Does anyone have any ideas to improve this?

**Comment by sebastian-sz** [comment]
I think we could consider running pytest in parallel, using [pytest-xdist](https://pytest-xdist.readthedocs.io/en/latest/).

Baseline: Current master branch + default command `pytest -v keras_cv/` takes 472.09s (0:07:52) on my machine.

Running 4 processes with `pytest-xdist` slices this time more than 2x: 
```
pip install pytest-xdist
pytest -v keras_cv -n 4  # 189.70s (0:03:09)
```
This argument can be probably tuned, depending on the machine.

This could be a starting point.

**Reply by bhack** [reply]
We could also evaluate to cluster tests in different jobs as jobs in GitHub Action runs in parallel.

We could also check if there are few tests with a very specific execution overhead.
